server/spec/auth/authorize_api_request_spec.rb

```diff
@@ -0,0 +1,73 @@
+# spec/auth/authorize_api_request_spec.rb
+require 'rails_helper'
+
+RSpec.describe AuthorizeApiRequest do
+  # Create test user
+  let(:user) { create(:api_user) }
+  # Mock `Authorization` header
+  let(:header) { { 'Authorization' => token_generator(user.id) } }
+  # Invalid request subject
+  subject(:invalid_request_obj) { described_class.new({}) }
+  # Valid request subject
+  subject(:request_obj) { described_class.new(header) }
+
+  # Test Suite for AuthorizeApiRequest#call
+  # This is our entry point into the service class
+  describe '#call' do
+    # returns user object when request is valid
+    context 'when valid request' do
+      it 'returns user object' do
+        result = request_obj.call
+        expect(result[:user]).to eq(user)
```

Comment:
```suggestion
        expect(result).to eq({ user: user })
```

By `expect`ing directly on a variable, a bad `result` (such as `nil`) would not break our expect reporting (by emitted a NoMethodError `#[]` on nil).

Reply:
updated as suggested